src/Aspire.Hosting.Azure.CognitiveServices/AzureOpenAIExtensions.cs

```diff
@@ -116,8 +100,44 @@ public static IResourceBuilder<AzureOpenAIResource> AddAzureOpenAI(this IDistrib
                 CognitiveServicesBuiltInRole.CognitiveServicesOpenAIContributor);
     }
 
+    private static CognitiveServicesAccountDeployment CreateDeployment(AzureResourceInfrastructure infrastructure, CognitiveServicesAccount cogServicesAccount, ref CognitiveServicesAccountDeployment? dependency, AzureOpenAIDeployment deployment)
+    {
+        var cdkDeployment = new CognitiveServicesAccountDeployment(Infrastructure.NormalizeBicepIdentifier(deployment.Name))
+        {
+            Name = deployment.Name,
+            Parent = cogServicesAccount,
+            Properties = new CognitiveServicesAccountDeploymentProperties()
+            {
+                Model = new CognitiveServicesAccountDeploymentModel()
+                {
+                    Name = deployment.ModelName,
+                    Version = deployment.ModelVersion,
+                    Format = "OpenAI"
+                }
+            },
+            Sku = new CognitiveServicesSku()
+            {
+                Name = deployment.SkuName,
+                Capacity = deployment.SkuCapacity
+            }
+        };
+        infrastructure.Add(cdkDeployment);
+
+        // Subsequent deployments need an explicit dependency on the previous one
+        // to ensure they are not created in parallel. This is equivalent to @batchSize(1)
+        // which can't be defined with the CDK
+
+        if (dependency != null)
+        {
+            cdkDeployment.DependsOn.Add(dependency);
+        }
+
+        dependency = cdkDeployment;
+        return dependency;
+    }
+
     /// <summary>
-    /// Adds an Azure OpenAI Deployment resource to the application model. This resource requires an <see cref="AzureOpenAIResource"/> to be added to the application model.
+    /// Adds an Azure OpenAI Deployment to the <see cref="AzureOpenAIResource"/> resource. This resource requires an <see cref="AzureOpenAIResource"/> to be added to the application model.
     /// </summary>
```

Comment:
Can we obsolete this one now or does it still server a purpose?

Reply:
That is a good topic for discussion. On one hand, we don't necessarily _have to_ obsolete it. But telling people "hey you should move from this old API to new AI" seems useful. Thoughts?